src/number/format.js

```diff
@@ -68,6 +68,29 @@ return function( number, properties ) {
 		number *= 1000;
 	}
 
+	var compactPattern, compactDigits, compactProperties, divisor, pluralForm, rounded;
+	if ( compactMap ) {
+		var zeroes = Array(Math.floor(number).toString().length).join( "0" );
+		if ( zeroes.length >= 3 ) {
+			// use default plural form to perform initial decimal shift
+			compactPattern = compactMap[ "1" + zeroes + "-count-other" ];
+			compactDigits = compactPattern.split( "0" ).length - 1;
+			divisor = zeroes.length - ( compactDigits - 1 );
+			number = number / Math.pow( 10, divisor );
+
+			// calculate rounded value to determine plural form and extract correct pattern
+			rounded = parseInt(numberFormatIntegerFractionDigits( number, minimumIntegerDigits,
+				minimumFractionDigits, maximumFractionDigits, round, roundIncrement ));
+			pluralForm = pluralGenerator ? pluralGenerator(rounded) : "other";
+			compactPattern = compactMap[ "1" + zeroes + "-count-" + pluralForm ] || compactPattern;
```

Comment:
Yeap, but in theory that should never happen, for example the Japanese example, `pluralGenerator` should never return `one`. Also, CLDR has all the necessary forms for a locale. For example, for cases like Arabic where there are various plural forms, compactMap should contain all of them.

 . What I mean is,